source/common/router/scoped_rds.cc

```diff
@@ -393,16 +393,13 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
 }
 
 void ScopedRdsConfigSubscription::onRdsConfigUpdate(const std::string& scope_name,
-                                                    RdsRouteConfigSubscription& rds_subscription) {
+                                                    ConfigConstSharedPtr new_rds_config) {
   auto iter = scoped_route_map_.find(scope_name);
   ASSERT(iter != scoped_route_map_.end(),
          fmt::format("trying to update route config for non-existing scope {}", scope_name));
   auto new_scoped_route_info = std::make_shared<ScopedRouteInfo>(
       envoy::config::route::v3::ScopedRouteConfiguration(iter->second->configProto()),
-      std::make_shared<ConfigImpl>(
-          rds_subscription.routeConfigUpdate()->protobufConfiguration(), optional_http_filters_,
-          factory_context_, factory_context_.messageValidationContext().dynamicValidationVisitor(),
```

Comment:
Can you confirm the Config doesn't hold a factory_context_ from another networkfilterchain/listener?
Reuse a protobuf nessage is pretty safe but Config is risk when the config is shared among different filter chain. There were a lot of issues

Reply:
I think they all use the factory_context_ of [ScopedRdsConfigSubscription.](https://github.com/envoyproxy/envoy/blob/6b18c923fae9d583d80de348f9656d7420ab8bb8/source/common/router/scoped_rds.cc#L194)

Reply:
LGTM. Thank you for hte explanation!